### PR TITLE
Handle additional error case

### PR DESCRIPTION
**Pull request by charlesbullis**
Gracefully handle case where fetch raises an exception with response undefined.

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amazon-cognito-identity-js/src/Client.js [hunk 1]
```diff
@@ -56,6 +56,13 @@ export default class Client {
         return callback(error);
       })
       .catch(() => {
+        if (!response) {
+          return callback({
+            code: 418,
```
Comment (veerabio) [inline packages/amazon-cognito-identity-js/src/Client.js]: is 418 appropriate for this?

Comment (charlesbullis) [reply packages/amazon-cognito-identity-js/src/Client.js]: No, it is not appropriate. Suggestions?

Comment (mlabieniec) [reply packages/amazon-cognito-identity-js/src/Client.js]: lol nice! :tea:  Perhaps just a simple 400 works?

**Comment by elorzafe** [comment]
@charlesbullis thanks for your PR, this issue was solve with PR #706

Thanks again!

**Pull request closed without merging**